platform/http_client_apple.mm

```diff
@@ -33,23 +33,68 @@ of this software and associated documentation files (the "Software"), to deal
 
 #include "base/logging.hpp"
 
+
+@interface RedirectDelegate : NSObject<NSURLSessionDataDelegate>
+
+@property(nonatomic) BOOL handleRedirects;
+
+- (instancetype)init:(BOOL)handleRedirects;
+
+- (void)        URLSession:(NSURLSession *)session
+                      task:(NSURLSessionTask *)task
+willPerformHTTPRedirection:(NSHTTPURLResponse *)response
+                newRequest:(NSURLRequest *)newRequest
+         completionHandler:(void (^)(NSURLRequest *))completionHandler;
+@end
+
+@implementation RedirectDelegate
+- (instancetype)init:(BOOL)handleRedirects
+{
+  self = [super init];
+  if (self)
+  {
+    _handleRedirects = handleRedirects;
+  }
+  
+  return self;
+}
+
+- (void)        URLSession:(NSURLSession *)session
+                      task:(NSURLSessionTask *)task
+willPerformHTTPRedirection:(NSHTTPURLResponse *)response
+                newRequest:(NSURLRequest *)newRequest
+         completionHandler:(void (^)(NSURLRequest *))completionHandler
+{
+    if (_handleRedirects && response.statusCode >= 300 && response.statusCode < 400) {
+        completionHandler(nil);
+    }
+    else {
+        completionHandler(newRequest);
+    }
+}
+@end
```

Comment:
Current implementation of `bool HttpClient::RunHttpRequest()` doesn't support handling of redirects.

**Actual behaviour:**

1. Make HTTP request to `osm.org/oauth2/authorize`
2. Server responses with HTTP Status 302 and redirects to `om://oauth2/osm/callback?code=xQpf58...`
3. iOS follow redirect and tries to open URL `om://oauth2/osm/callback?code=xQpf58...`
4. iOS gives error

**Expected behaviour:**
1. Make HTTP request to `osm.org/oauth2/authorize`
2. Server responses with HTTP Status 302 and redirects to `om://oauth2/osm/callback?code=xQpf58...`
3. If `HttpClient::handle_redirects` field is `true` then do not follow redirect and give back response from server.

To handle redirect I had to implement `RedirectDelegate`. Is it made correctly?

Reply:
Since when did increasing the entropy become a better solution? ;)

Reply:
Replaced `handleRedirects` with `followRedirects` all over the code.